RocketChat/Rocket.Chat.ReactNative
# feat: new audio player

## Proposed changes

This PR is improving the Audio Player. The main reason was to change the layout and add the capability to increase/decrease the audio's velocity (rate).

Beyond the layout stuff, I changed how we handle the audio. Also, I refactor the approach implemented in that PR https://github.com/RocketChat/Rocket.Chat.ReactNative/pull/4662 to avoid the necessity of EventListener. Added to this new approach, we are able to control and unload the audio from other places besides the RoomView and the Component Audio, this could be discussed later.

## Issue(s)	


## How to test or reproduce

- Send multiple audios in a single channel to test the play/pause/rate
- Send audio in a thread 
    - The audio from a channel will pause when opening the thread
    - If the audio from a thread is playing, then back to the channel, this audio from the thread should stop
 - Background listening
 - Check if screen is always on while playing an audio and if can go off after is paused
 - Play the audio from all clients origin on all clients Android/iOS.

## Screenshots

https://github.com/RocketChat/Rocket.Chat.ReactNative/assets/47038980/87c08f06-d281-40b6-990f-d56ce67d625e


## Types of changes




## Checklist



## Further comments

[TC-781]

[TC-781]: https://rocketchat.atlassian.net/browse/TC-781?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
